internal/common/test_configuration.go

```diff
@@ -97,6 +98,50 @@ func GenerateTestDefaultInventoryIPv4Only() string {
 	Expect(err).To(Not(HaveOccurred()))
 	return string(b)
 }
+func GenerateTestDefaultInventoryWithNetworkInterface(interfaces []*models.Interface, hostname string) string {
```

Comment:
At the end I resolved into merging my `GenerateTestDefaultInventory*` methods and the `GenerateTestDefaultInventoryIPv4Only` into one `GenerateTestInventoryWithNetwork` where you can pass a small structure that contains the IPv4, IPv6 and hostname values. I debated whether to pass the `Interface` structure directly for IPs and then hostname as a separated argument, but I figured that it might add more complexity by having to define the extra arguments for the `Interface`, such as `MACAddress` or `Name`, when they are actually not used in the tests.
https://github.com/openshift/assisted-service/blob/7ac5f5d8ef663c66f9d5a1613a12a73da688c101/internal/common/test_configuration.go#L91-L95

So now there are 2 methods to generate the inventory:
* `GenerateTestDefaultInventory()`
* `GenerateTestInventoryWithNetwork(netAddress NetAddress)`